source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1857,10 +1874,19 @@ Http::ConnectionPool::InstancePtr ProdClusterManagerFactory::allocateConnPool(
 #endif
   }
   if (protocols.size() >= 2) {
+    ENVOY_BUG(origin.has_value(), "Unable to determine origin for host ");
```

Comment:
Should we put this inside the feature guard?

Reply:
done